types/src/v2/params.rs

```diff
@@ -167,23 +167,28 @@ impl<'a> RpcParams<'a> {
 	{
 		self.parse::<[T; 1]>().map(|[res]| res)
 	}
-
-	/// Creates an owned version of parameters.
-	/// Required to simplify proc-macro implementation.
-	#[doc(hidden)]
-	pub fn owned(self) -> OwnedRpcParams {
-		self.into()
-	}
 }
 
 /// Owned version of [`RpcParams`].
 #[derive(Clone, Debug)]
 pub struct OwnedRpcParams(Option<String>);
 
 impl OwnedRpcParams {
-	/// Converts `OwnedRpcParams` into borrowed [`RpcParams`].
-	pub fn borrowed(&self) -> RpcParams<'_> {
-		RpcParams(self.0.as_ref().map(|s| s.as_ref()))
+	/// Attempt to parse all parameters as array or map into type `T`
+	pub fn parse<T>(self) -> Result<T, CallError>
```

Comment:
The reason this is a draft is that I'm not sure if there's a better way to do this than duplicating code from `RpcParams`.